examples/01_standalone_sdk/02_custom_tools.py

```diff
@@ -131,14 +131,14 @@ def __call__(self, action: GrepAction, conversation=None) -> GrepObservation:  #
 cwd = os.getcwd()
 
 
-def _make_bash_and_grep_tools(conv_state) -> list[ToolDefinition]:
+def _make_bash_and_grep_tools(conv_state) -> list[ToolBase]:
     """Create execute_bash and custom grep tools sharing one executor."""
 
     bash_executor = BashExecutor(working_dir=conv_state.workspace.working_dir)
     bash_tool = execute_bash_tool.set_executor(executor=bash_executor)
 
     grep_executor = GrepExecutor(bash_executor)
-    grep_tool = ToolDefinition(
+    grep_tool = ToolBase(
```

Comment:
@OpenHands, is there any way to stop instantiating `ToolBase`? It feels wrong to initialize a base class. How about we change the way we define new tool by forcing them to define a new GlobTool and directly instantiate that